solana/solana-ibc/programs/solana-ibc/src/lib.rs

```diff
@@ -169,6 +168,23 @@ pub mod solana_ibc {
             .map_err(move |err| error!((&err)))
     }
 
+    /// Forms the message from the data stored in the chunks account
+    ///
+    /// Since the instructions can only be of maximum 1232 bytes, for large messages,
+    /// we need to split them into chunks and store them in accounts ( who have upto 10Mb)  
+    /// of storage. We can then deserialize the message and send it to ibc-rs.
+    pub fn deliver_with_chunks<'a, 'info>(
+        ctx: Context<'a, 'a, 'a, 'info, DeliverWithChunks<'info>>,
+    ) -> Result<()> {
+        let chunks = ctx.accounts.chunks.try_borrow_data().unwrap();
+        let message = ibc::MsgEnvelope::try_from_slice(&chunks).unwrap();
```

Comment:
I suggest we require chunks account to be length-prefixed, so
something like:

```rust
    let chunks = std::split_at::<4>(chunks).and_then(|(len, chunks)| {
        chunks.get(..u32::from_be_bytes(len) as usize)
    }).ok_or(...)?;
    let message = ...;
```

This will make sure we don’t read some garbage data left within the
account if it has been reused.

Reply:
sure sure, makes sense.